### ISSUE TITLE
scrape_text_by_id will not find IDs that are not in double quotes

### ISSUE THREAD

**Issue by petdance** [petdance/test-www-mechanize#47]
There's an optimization to not parse any HTML if the ID is not in the document somewhere.

````
my $found = index( $html, "id=\"$id\"" );
if ( $found >= 0 ) {
    my $parser = HTML::TokeParser->new( \$html );
````

This is incomplete.  We have to pass all sorts of cases like

````
id="foo"
id='foo'
id = 'foo'
id = foo
````

etc etc etc.  Make this optimization more robust, or remove it entirely.

**Comment by petdance** [comment]
Fixed in 1.45_04.

**Comment by petdance** [comment]
4cd8294c8f972fa575d0cea9b46c4d512cdff86e
